packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2930,6 +2932,33 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return expr;
       }, state);
 
+      // report error if single type parameter used without trailing comma.
+      if (
+        this.hasPlugin("jsx") &&
+        arrow.node?.type === "ArrowFunctionExpression" &&
+        arrow.node.typeParameters?.params.length === 1 &&
+        !arrow.node.typeParameters.extra?.trailingComma
+      ) {
+        const parameter = arrow.node.typeParameters.params[0];
+        if (parameter.constraint) {
+          // If Type has any constraints, means that it is not alone their.
+          // <T extends U> is a valid declaration.
+          // <T extends {name: string}> is also a valid declaration.
+        } else {
+          const node = parameter.name;
+          let identifierName = "";
+          if (typeof node === "object") identifierName = node.name;
+          else identifierName = node;
```

Comment:
```suggestion
          const identifierName = process.env.BABEL_8_BREAKING ? parameter.name.name : parameter.name;
```

So we can remove Babel 7 AST support when we release Babel 8.